File: typecast-php/src/Models/RecommendedVoice.php
<?php

declare(strict_types=1);

namespace Neosapience\Typecast\Models;

/**
 * Voice recommendation result.
 *
 * Recommendation results only include the matched voice ID, voice name, and
 * similarity score. Use getVoiceV2() or getVoicesV2() to fetch detailed voice
 * metadata for a returned voice ID.
 */
class RecommendedVoice
{
    public function __construct(
        public string $voiceId,
        public string $voiceName,
        public float $score,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        return new self(
            voiceId: $data['voice_id'] ?? '',
            voiceName: $data['voice_name'] ?? '',
            score: (float) ($data['score'] ?? 0),
        );
    }
}

File: typecast-java/src/main/java/com/neosapience/models/RecommendedVoice.java
package com.neosapience.models;

/**
 * Voice recommendation result.
 *
 * <p>Recommendation results only include the matched voice ID, voice name, and
 * similarity score. Use {@code getVoiceV2} or {@code getVoicesV2} to fetch
 * detailed voice metadata for a returned voice ID.</p>
 */
public class RecommendedVoice {
    private String voiceId;
    private String voiceName;
    private double score;

    public RecommendedVoice() {
    }

    public RecommendedVoice(String voiceId, String voiceName, double score) {
        this.voiceId = voiceId;
        this.voiceName = voiceName;
        this.score = score;
    }

    public String getVoiceId() {
        return voiceId;
    }

    public void setVoiceId(String voiceId) {
        this.voiceId = voiceId;
    }

    public String getVoiceName() {
        return voiceName;
    }

    public void setVoiceName(String voiceName) {
        this.voiceName = voiceName;
    }

    public double getScore() {
        return score;
    }

    public void setScore(double score) {
        this.score = score;
    }
}
